cartridges/int_brain_commerce/cartridge/scripts/services/rezolveSnpdService.js

```
@@ -0,0 +1,242 @@
+        return LocalServiceRegistry.createService('rezolvesnpd.http.ingest', {
+            createRequest: function (svc, params) {
+                const baseUrl = Site.current.getCustomPreferenceValue('rezolveIngestionApiUrl');
+                const endPointPath = params && params.endPointConfigs && params.endPointConfigs.endPoint ? params.endPointConfigs.endPoint : '';
+                const fullUrl = baseUrl + (endPointPath || '');
+                svc.setURL(fullUrl);
+                const method = params && params.endPointConfigs && params.endPointConfigs.method ? params.endPointConfigs.method : 'POST';
+                svc.setRequestMethod(method);
+
+                // Set headers according to expected API contract
+                if (Site.current.getCustomPreferenceValue('rezolveClientKey')) {
+                    svc.addHeader('Authorization', 'client-key ' + Site.current.getCustomPreferenceValue('rezolveClientKey'));
+                }
+                if (Site.current.getCustomPreferenceValue('rezolveCustomerId')) {
+                    svc.addHeader('X-Groupby-Customer-Id', Site.current.getCustomPreferenceValue('rezolveCustomerId'));
+                }
+
+                if (method === 'GET') {
+                    svc.addHeader('Accept', 'application/json');
+                    return null;
+                }
+
+                const fields = params && params.requestBody ? params.requestBody : {};
+                const requestParts = [];
+
+                Object.keys(fields).forEach(function (key) {
+                    if (key === 'catalog') {
+                        if (typeof fields[key] === 'string') {
+                            const file = new File(fields[key]);
+                            if (file.exists()) {
+                                requestParts.push(new HTTPRequestPart(key, file));
+                                Logger.info('Added file part for catalog: {0}', file.getFullPath());
```

If the file doesn't exist we fallback to logging full paths as plain text which can log implementation details and sending an invalid file path string might not be accepted by the API.

We can log only file name at `debug`:
` Logger.debug('Added file part for catalog: {0}', file.getName());`

We should fast fail if the file is missing instead of silently sending text.

---

Fixed